src/test/ui/existential-type/issue-52843-closure-constrain.rs

```diff
@@ -0,0 +1,12 @@
+// Checks to ensure that we properly detect when a closure constrains an existential type
+#![feature(existential_type)]
+
+use std::fmt::Debug;
+
+fn main() {
+    existential type Existential: Debug;
+    fn _unused() -> Existential { String::new() }
+    //~^ ERROR: concrete type differs from previous defining existential type use
```

Comment:
moderately confusing to have the first item (textually) talk about "previous". Maybe we should reword "previous" to "other"